apps/condo/domains/ticket/components/TicketChanges/TicketChange.tsx

```diff
@@ -0,0 +1,178 @@
+    const UnitNameMessage = intl.formatMessage({ id: 'pages.condo.ticket.TicketChanges.unitName' })
+    const UnitNameChangedMessage = intl.formatMessage({ id: 'pages.condo.ticket.TicketChanges.unitName.change' })
+    const AssigneeMessage = intl.formatMessage({ id: 'pages.condo.ticket.TicketChanges.assignee' })
+    const ClassifierMessage = intl.formatMessage({ id: 'pages.condo.ticket.TicketChanges.classifier' })
+    const fields = [
+        ['clientPhone', ClientPhoneMessage],
+        ['details', DetailsMessage],
+        ['clientName', ClientNameMessage],
+        ['isPaid', IsPaidMessage],
+        ['isEmergency', IsEmergencyMessage],
+        ['statusDisplayName', StatusDisplayNameMessage],
+        ['unitName', UnitNameMessage],
+        ['assigneeDisplayName', AssigneeMessage],
+        ['classifierDisplayName', ClassifierMessage],
+    ]
+
+    const BooleanToString = {
+        isPaid: {
+            'true': intl.formatMessage({ id: 'pages.condo.ticket.TicketChanges.isPaid.true' }),
+            'false': intl.formatMessage({ id: 'pages.condo.ticket.TicketChanges.isPaid.false' }),
+        },
+        isEmergency: {
+            'true': intl.formatMessage({ id: 'pages.condo.ticket.TicketChanges.isEmergency.true' }),
+            'false': intl.formatMessage({ id: 'pages.condo.ticket.TicketChanges.isEmergency.false' }),
+        },
+    }
+
+    const formatField = (field, value) => {
+        const formatterFor = {
+            clientPhone: (field, value) => (
+                <PhoneLink value={value}/>
+            ),
+            details: (field, value) => (
+                value.length > 30 ? value.slice(0, 30) + '…' : value
+            ),
+            unitName: (field, value) => (
+                // Formally, unit name (e.g. apartment) was changed, but semantically,
+                // was changed a whole address of the ticket, so, to preserve context,
+                // the change of the unit is displayed as the change of the address
+                UnitNameChangedMessage
+                    .replace('{address}', ticketChange.ticket.property.address)
+                    .replace('{unitName}', value)
+            ),
+        }
+        return has(formatterFor, field)
+            ? formatterFor[field](field, value)
+            : value
+    }
+
+    const format = (field, value) => (
+        typeof value === 'boolean'
+            ? BooleanToString[field][value]
+            : formatField(field, value)
+    )
+
+    /*
+        Interpolates message string with JSX tags.
+        They will be safely mounted in place of `{to}` and `{from}` placeholders
+     */
+    const formatDiffMessage = (field, message, ticketChange) => {
+        // we have both "from" and "to" parts to interpolate
+        if (message.search('{from}') !== -1 && message.search('{to}') !== -1) {
+            const parts1 = message.split('{from}')
```

Comment:
hat is parts1? add readable variable names

Reply:
Renamed in `827d9982`